src/nprintml/label_aggregation.py

```diff
@@ -0,0 +1,233 @@
+
+        print('Loading nprint')
+        npt_df = self.load_npt(self.npts)
+        print('  nPrint shape: {0}'.format(npt_df.shape))
+        if compress:
+            print('Compressing nPrint')
+            npt_df = self.compress_npt(npt_df)
+            print('  compressed nPrint shape: {0}'.format(npt_df.shape))
+
+        print('Loading labels')
+        labels = self.load_labels(self.label_file)
+        print('  number of labels: {0}'.format(labels.shape[0]))
+
+        if sample_size > 1:
+            print('Grouping by sample size')
+            npt_df = self.group_by_sample_size(npt_df, sample_size)
+            print('  New shape of dataframe: {0}'.format(npt_df.shape))
+
+        print('Attaching labels to nPrints')
+        npt_df, missing_labels, ogns, nns = self.attach_labels(npt_df, labels)
+        print('  labels attached: missing labels for:')
+        for missing_label in missing_labels:
+            print('    {0}'.format(missing_label))
+        print('    missing labels caused {0} samples to be dropped'.format(ogns - nns))
+
+        return npt_df
+
+    def group_by_sample_size(self, npt, sample_size):
+        '''
+        Takes a large nPrint dataframe, groups it by the index, then splits
+        each separated sample into n sample_sized samples
+        '''
+        # If the sample size is 1, its just per packet, which we already have
+        if sample_size <= 1:
+            return npt
+
+        new_data = []
+        indexes = []
+        new_columns = self.get_flattened_columns(npt, sample_size)
+        groups = npt.groupby(npt.index)
+        for lid, group in groups:
+            # Pad rows to get an even division for sample size
+            pad_rows = sample_size - group.shape[0] % sample_size
+            fill_row = [-1 for i in range(group.shape[1])]
+            for i in range(group.shape[0], group.shape[0] + pad_rows):
+                group.loc[i] = fill_row
+
+            # actually separate group into samples and flatten for 1d learning
+            num_splits = group.shape[0] / sample_size
+            for sample in np.split(group, num_splits):
+                flattened_sample = sample.to_numpy().flatten()
+                new_data.append(flattened_sample)
+                indexes.append(lid)
+
+        new_df = pd.DataFrame(new_data, index=indexes, columns=new_columns)
+
+        return new_df
+
+
+class PcapLabelAggregator(LabelAggregator):
+    '''
+    PcapFeatureAggregtor is meant to take a directory of nPrint files and consider
+    each file as a single sample, padding to the maximum size and running autoML
+    on the generated samples
+    '''
+    def generate_features(self, compress=False, sample_size=1):
+        '''
+        Enumerates the directory of nPrints given, loading them, padding to
+        their maximum size, and attaching labels
+        '''
+        npt_dir = pathlib.Path(self.npts)
+        files = [npt_f for npt_f in npt_dir.glob('**/*') if npt_f.is_file()]
```

Comment:
Is the idea that the nprint results may be at any depth (not just top-level in the directory)? Will that happen?